src/s2geography/geoarrow.cc

```diff
@@ -742,6 +743,249 @@ void Reader::ReadGeography(const ArrowArray* array, int64_t offset,
+      coords_view_.values[1] = &coords[1];
+      // coords_view_.values = static_cast<const double*[4]>(coords);
+      // GEOARROW_COORD_VIEW_VALUE(&coords_view_, 0, 0) = ll.lng().degrees();
+      // GEOARROW_COORD_VIEW_VALUE(&coords_view_, 0, 1) = ll.lng().degrees();
+
+      GEOARROW_RETURN_NOT_OK(visitor_.coords(&visitor_, &coords_view_));
+      GEOARROW_RETURN_NOT_OK(visitor_.geom_end(&visitor_));
+    }
+    // } else {
+    //   handler->new_geometry_type(util::GeometryType::MULTIPOINT);
+    //   HANDLE_OR_RETURN(handler->geom_start(util::GeometryType::MULTIPOINT,
+    //                                        geog.Points().size()));
+
+    //   for (const S2Point& pt : geog.Points()) {
+    //     handler->geom_start(util::GeometryType::POINT, 1);
+    //     S2LatLng ll(pt);
+    //     coords[0] = ll.lng().degrees();
+    //     coords[1] = ll.lat().degrees();
+    //     HANDLE_OR_RETURN(handler->coords(coords, 1, 2));
+    //     HANDLE_OR_RETURN(handler->geom_end());
+    //   }
+
+    //   handler->geom_end();
+    // }
+    std::cout << "-- end of VisitPoints" << "\n";
+    return GEOARROW_OK;
+  }
+
+  int VisitFeature(const Geography& geog) {
+    std::cout << "In VisitFeature" << "\n";
+    GEOARROW_RETURN_NOT_OK(visitor_.feat_start(&visitor_));
+
+    auto child_point = dynamic_cast<const PointGeography*>(&geog);
+    if (child_point != nullptr) {
+      GEOARROW_RETURN_NOT_OK(VisitPoints(*child_point));
+    } else {
+      throw Exception("Unsupported Geography subclass");
+      // auto child_polyline = dynamic_cast<const PolylineGeography*>(&geog);
+      // if (child_polyline != nullptr) {
+      //   HANDLE_OR_RETURN(handle_polylines(*child_polyline, handler));
+      // } else {
+      //   auto child_polygon = dynamic_cast<const PolygonGeography*>(&geog);
+      //   if (child_polygon != nullptr) {
+      //     HANDLE_OR_RETURN(handle_polygon(*child_polygon, handler));
+      //   } else {
+      //     auto child_collection = dynamic_cast<const GeographyCollection*>(&geog);
+      //     if (child_collection != nullptr) {
+      //       HANDLE_OR_RETURN(handle_collection(*child_collection, handler));
+      //     } else {
+      //       throw Exception("Unsupported Geography subclass");
+      //     }
+      //   }
+      // }
+    }
+    std::cout << "-- end of VisitFeature" << "\n";
+    return GEOARROW_OK;
+  }
+
+  void ThrowNotOk(int code) {
+    if (code != GEOARROW_OK) {
+      throw Exception(error_.message);
+    }
+  }
+};
+
+
+Writer::Writer() : impl_(new WriterImpl()) {}
+
+Writer::~Writer() { impl_.reset(); }
+
+void Writer::Init(const ArrowSchema* schema, const ImportOptions& options) {
+  impl_->Init(schema, options);
+}
+
+void Writer::Init(OutputType output_type, const ImportOptions& options,  struct ArrowSchema* out_schema) {
+  switch (output_type) {
+    case OutputType::kPoints:
+      impl_->Init(GEOARROW_TYPE_INTERLEAVED_POINT, options, out_schema);
+      break;
```

Comment:
I just added `kPoints` (mapping to interleaved points) for a quick test. I assume we might want to add some generic "native geoarrow" option like `kGeoArrow(Interleaved)`, and then infer the schema from the geographies? (with something similar like `GeoArrowGEOSSchemaCalculator` you did in geoarrow-c-geos)

It might still be worth allowing the user to specify the type themselves to avoid inference / choose a specific type (e.g. always the multi-version regardless of presence of any multi geom)